### PR TITLE
refactor(*): De-angularize caches

### PR DESCRIPTION
**Pull request by jrsquared**

### CODE REVIEW

File: app/scripts/modules/core/src/cache/collapsibleSectionStateCache.ts [hunk 1]
```diff
@@ -35,8 +32,4 @@ export class CollapsibleSectionStateCache {
   }
 }
 
-export const COLLAPSIBLE_SECTION_STATE_CACHE = 'spinnaker.core.cache.collapsibleSectionState';
-module(COLLAPSIBLE_SECTION_STATE_CACHE, [require('angular-cache')]).service(
-  'collapsibleSectionStateCache',
-  CollapsibleSectionStateCache,
-);
+export let CollapsibleSectionStateCache = new CollapsibleSectionStateCacheInternal();
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/cache/collapsibleSectionStateCache.ts]: const?

File: app/scripts/modules/core/src/projects/projects.controller.spec.js [hunk 2]
```diff
@@ -40,6 +40,7 @@ describe('Controller: Projects', function() {
           $timeout: $timeout,
         });
 
+        this.$scope.viewState.projectFilter = '';
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/projects/projects.controller.spec.js]: huh

Comment (anotherchrisberry) [reply app/scripts/modules/core/src/projects/projects.controller.spec.js]: i mean sure

**Comment by anotherchrisberry** [comment]
LGTM - thanks for doing this

File: app/scripts/modules/core/src/cache/deckCacheFactory.ts [hunk 3]
```diff
@@ -147,35 +127,35 @@ export class DeckCacheService {
...
         cacheFactory.get(key).removeAll();
         cacheFactory.get(key).destroy();
       }
     }
   }
 
-  private static getStats(cache: ICache): IStats {
-    const keys: string[] = cache.keys();
+  private static getStats(cache: Cache): IStats {
+    const keys: Array<string | number> = cache.keys();
```
[the remark this answers is not in the document]
Comment (jrsquared) [reply app/scripts/modules/core/src/cache/deckCacheFactory.ts]: cache.keys() is already an `Array<string | number>`, but I guess I could cast it. Either way, I shouldn't have put the type def there like an idiot.

**Pull request merged**